File: djello/db/seeds.rb
# This file should contain all the record creation needed to seed the database with its default values.
# The data can then be loaded with the rake db:seed (or created alongside the db with db:setup).
#
# Examples:
#
#   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
#   Mayor.create(name: 'Emanuel', city: cities.first)


User.destroy_all
Board.destroy_all
List.destroy_all
Card.destroy_all
CardMembership.destroy_all

#create
2.times do

  user = User.create(username: Faker::Internet.user_name,
    email:    Faker::Internet.email,
    password: 'foobar_1234',
    password_confirmation: 'foobar_1234')
  3.times do |i|
    board = user.boards.create(title: "Board Title #{i}")

    2.times do
      list = board.lists.create(title: Faker::Commerce.department,
        description:  Faker::Lorem.sentence)

      4.times do
        list.cards.create(title: Faker::SlackEmoji.activity,
          description: Faker::Lorem.sentence )
      end
    end
  end
end

testuser = User.create(username: 'tester',
  email:    Faker::Internet.email,
  password: 'password',
  password_confirmation: 'password')

3.times do |i|
  board = testuser.boards.create(title: "Random Board Title #{i}")
  2.times do
    list = board.lists.create(title: Faker::Commerce.department,
      description:  Faker::Lorem.sentence )
    4.times do
      list.cards.create(title: Faker::SlackEmoji.activity,
        description: Faker::Lorem.sentence )
    end
  end
end

users = User.all
cards = Card.all

25.times do

  CardMembership.create(user_id: users.sample.id,
                        card_id: cards.sample.id)
end

File: server/seeds.js
const models = require("./models");
const faker = require("faker");
const voca = require("voca");
const mongooseeder = require("mongooseeder");

let { User, Board, List, Card } = models;

function seeds() {
  let users = [];
  for (let i = 0; i < 5; i++) {
    let user = new User({
      username: faker.random
        .words(2)
        .split(" ")
        .join(""),
      password: "pass"
    });
    users.push(user);
  }

  let boards = [];
  for (let i = 0; i < 20; i++) {
    let board = new Board({
      title: voca.titleCase(faker.random.words(4)),
      user: users[i % 10]
    });
    users[i % 5].boards.push(board);
    board.members.push(users[i % 5]);
    boards.push(board);
  }

  let lists = [];
  for (let i = 0; i < 60; i++) {
    let list = new List({
      title: voca.titleCase(faker.random.words(4)),
      description: faker.random.words(10),
      board: boards[i % 20]
    });
    boards[i % 20].lists.push(list);
    lists.push(list);
  }

  let cards = [];
  for (let i = 0; i < 120; i++) {
    let card = new Card({
      title: voca.titleCase(faker.random.words(4)),
      body: faker.random.words(20),
      list: lists[i % 60],
      members: [users[i % 5]]
    });
    lists[i % 60].cards.push(card);
    users[i % 5].cards.push(card);
    cards.push(card);
  }

  const promises = [users, boards, lists, cards].reduce(
    (ps, collection) => ps.concat(collection.map(model => model.save())),
    []
  );

  return Promise.all(promises);
}

mongooseeder.seed({
  mongodbUrl: require("./config/mongoUrl"),
  models: models,
  clean: true,
  mongoose: require("mongoose"),
  seeds: seeds
});
